### PR TITLE
add initial clickhouse support

### PR DESCRIPTION
**Pull request by ne1r0n**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+base.ischema_names["Date32"] = DATE
+base.ischema_names["Bool"] = BOOLEAN
+base.ischema_names["Nothing"] = sqltypes.NullType
+base.ischema_names["Int128"] = INTEGER
+base.ischema_names["Int256"] = INTEGER
+base.ischema_names["UInt128"] = INTEGER
+base.ischema_names["UInt256"] = INTEGER
+
+register_custom_type(custom_types.common.Array, ArrayTypeClass)
+register_custom_type(custom_types.ip.IPv4, NumberTypeClass)
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:67]: IPv6 is mapped as string but this one as Number, why?

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+                    dataset_properties_aspect.customProperties = custom_properties
+                    dataset_snapshot.aspects = aspects
+
+                    dataset_snapshot.aspects.append(dataset_properties_aspect)
+
+            # Emit the work unit from super.
+            yield wu
+
+    def _get_db_name(self) -> str:
+        return getattr(self.config, "database_alias")
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:382]: Is database alias a mandatory config parameter? 
As I can see there is a database property as well. 
Is it possible to set only database and not database_alias?

**Review by treff7es — COMMENTED** [review]
Awesome work, thank you.
I left a few minor comments.
The build is currently failing because I think you should add ClickHouse as dependency for the integration tests in setup.py.

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py (hunk 1 above)
Comment (ne1r0n) [reply metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:67]: In ClickHouse those two IP types are based on different data type.
More info in the documentation [IPv4](https://clickhouse.com/docs/en/sql-reference/data-types/domains/ipv4/) and [IPv6](https://clickhouse.com/docs/en/sql-reference/data-types/domains/ipv6/).

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py (hunk 2 above)
Comment (ne1r0n) [reply metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:382]: `database` property is used only for the connection.
`database_alias` is used as `platform_instance` and most likely should be replaced to it.
Both parameters are not mandatory.

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py (hunk 1 above)
Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:67]: thanks for the clarificaiton

File: metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py (hunk 2 above)
Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py:382]: @ne1r0n : I would recommend using the `platform_instance` config going forward and not having `database_alias` option in this connector, now that we have platform instance support in the core classes.

Comment (ne1r0n) [reply metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py]: @shirshanka I've changed `database_alias` to `platform_instance`

**Review by shirshanka — APPROVED** [review]
LGTM! Will merge after CI is green.

**Pull request merged**